x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/QueryPlan.java

```diff
@@ -140,11 +140,13 @@ private static Object doTransformExpression(Object arg, Function<Expression, ? e
         // preserving the type information is hacky and weird (a lot of context needs to be passed around and the lambda itself
         // has no type info so it's difficult to have automatic checking without having base classes).
 
-        if (arg instanceof Collection<?> c) {
+        if (arg instanceof List<?> c) {
```

Comment:
Please confirm we do not expect sets in here.

Reply:
We don't however it's not enforced - add an else throwing an error for safety.